src/server/scripts/Kalimdor/HallsOfOrigination/halls_of_origination.cpp

```diff
@@ -0,0 +1,251 @@
+    EVENT_FOLLOW_SUMMONER,
+};
+
+// 39612 - Spatial Flux (trash)
+class npc_hoo_spatial_flux : public CreatureScript
+{
+public:
+    npc_hoo_spatial_flux() : CreatureScript("npc_hoo_spatial_flux") { }
+
+    struct npc_hoo_spatial_fluxAI : public ScriptedAI
+    {
+        npc_hoo_spatial_fluxAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void Reset() override
+        {
+            events.Reset();
+            events.ScheduleEvent(EVENT_DUMMY_NUKE, Seconds(0));
+            events.ScheduleEvent(EVENT_SPAWN_ENERGY_FLUX, Seconds(3));
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+            
+            events.Update(diff);
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+
+            while (uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_DUMMY_NUKE:
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 100, true))
+                            DoCast(target, SPELL_DUMMY_NUKE);
+                        events.Repeat(Seconds(1));
+                        break;
+                    case EVENT_SPAWN_ENERGY_FLUX:
+                        DoCastSelf(SPELL_SPAWN_ENERGY_FLUX_TRASH);
+                        events.Repeat(Seconds(12));
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetHallsOfOriginationAI<npc_hoo_spatial_fluxAI>(creature);
+    }
+};
+
+// 44015 - Energy flux (trash)
+class npc_hoo_energy_flux : public CreatureScript
+{
+public:
+    npc_hoo_energy_flux() : CreatureScript("npc_hoo_energy_flux") { }
+
+    struct npc_hoo_energy_fluxAI : public ScriptedAI
+    {
+        npc_hoo_energy_fluxAI(Creature* creature) : ScriptedAI(creature)
+        {
+            me->SetReactState(REACT_PASSIVE);
+            target = nullptr;
+        }
+
+        void Reset() override
+        {
+            DoCastSelf(SPELL_ENERGY_FLUX_BEAM_TRASH);
+            DoCastSelf(SPELL_ENERGY_FLUX_PERIODIC);
+            me->SetWalk(true);
+            events.ScheduleEvent(EVENT_FOLLOW_SUMMONER, Seconds(1));
+            me->DespawnOrUnsummon(Seconds(6));
+        }
+
+        void IsSummonedBy(Unit* summoner) override
+        {
+            target = summoner;
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!target)
+                return;
+
+            events.Update(diff);
+
+            while (uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_FOLLOW_SUMMONER:
+                        me->GetMotionMaster()->MovePoint(0, target->GetPosition(), true);
```

Comment:
Why not MoveChase?

Reply:
Need to follow with zero distance, I don't think any other movement worked.

Reply:
https://hastebin.com/aficezomib.vbs